tests/src/ut_test_control_plane.c

```diff
@@ -0,0 +1,169 @@
+#include <unistd.h>
+
+/* Module Includes */
+#include <ut.h>
+#include <ut_log.h>
+
+#include <ut_control_plane.h>
+
+static UT_test_suite_t *gpAssertSuite1 = NULL;
+static UT_test_suite_t *gpAssertSuite2 = NULL;
+
+static ut_controlPlane_instance_t *gInstance = NULL;
+
+void wait_for_1_second()
+{
+    sleep(1);
+}
+
+void test_ut_control_testInitExit()
+{
+    ut_controlPlane_instance_t *pInstance = NULL;
+    ut_controlPlane_instance_t *pInstance1 = NULL;
+
+    UT_ControlPlane_Exit(NULL);
+    UT_ASSERT( pInstance == NULL );
+
+    pInstance = UT_ControlPlane_Init(8080);
+    UT_ControlPlane_Start(pInstance);
+    UT_ASSERT( pInstance != NULL );
+
+    wait_for_1_second();
+
+    pInstance1 = UT_ControlPlane_Init(9000);
+    UT_ControlPlane_Start(pInstance1);
+    UT_ASSERT( pInstance1 != NULL);
+
+    UT_ASSERT( pInstance != pInstance1 );
+    wait_for_1_second();
+
+    UT_ControlPlane_Exit(pInstance);
+    wait_for_1_second();
+
+    UT_ControlPlane_Exit(pInstance1);
+    wait_for_1_second();
+
+}
+
+void test_ut_control_testService()
+{
+    ut_controlPlane_instance_t *pInstance = NULL;
+    ut_controlPlane_instance_t *pInstance1 = NULL;
+
+    pInstance = UT_ControlPlane_Init(8080);
+    UT_ControlPlane_Start(NULL);
+    UT_ControlPlane_Start(pInstance);
+
+    wait_for_1_second();
+    pInstance1 = UT_ControlPlane_Init(9000);
+    UT_ControlPlane_Start(pInstance1);
+
+    wait_for_1_second();
+    UT_ASSERT( pInstance != pInstance1 );
+
+    UT_ControlPlane_Exit(pInstance);
+    UT_ASSERT( pInstance != pInstance1 );
+
+    UT_ASSERT( pInstance != NULL );
+    wait_for_1_second();
+
+    UT_ControlPlane_Exit(pInstance1);
+    UT_ASSERT( pInstance != pInstance1 );
+}
+
+void testCallback(char *key, ut_kvp_instance_t *instance)
+{
+    printf("*******************************Inside testCallback************************\n");
+    ut_kvp_print( instance );
+}
+
+void testRMFCallback(char *key, ut_kvp_instance_t *instance)
+{
+    UT_LOG("\n**************testRMFCallback is called****************\n");
+    ut_kvp_print( instance );
+}
+
+static void UT_ControlPlane_Sigint_Handler(int sig)
+{
+    wait_for_1_second();
+    UT_ControlPlane_Exit(gInstance);
+}
+
+void test_ut_control_performInit( void )
+{
+    gInstance = UT_ControlPlane_Init(8080);
+    UT_ASSERT(gInstance != NULL);
+
+    signal(SIGINT, UT_ControlPlane_Sigint_Handler);
+}
+
+void test_ut_control_performStart()
+{
+
+    UT_ControlPlane_RegisterCallbackOnMessage(gInstance, "test/myCommand", &testCallback);
+    UT_LOG("\n Please run this command from the command line");
+    UT_LOG("\n Waiting client test/myCommand to send yaml/json file");
+
+    UT_ControlPlane_RegisterCallbackOnMessage(gInstance, "test2/myCommand1", &testRMFCallback);
+    UT_LOG("\nWaiting client test/myCommand2 to send yaml/json file");
+
+    UT_ControlPlane_Start(gInstance);
+}
+
+void wait_function()
+{
+    for(int i = 0; i < 20; i++)
+    {
+        UT_LOG("Waiting for %d sec", i);
+        sleep(1);
```

Comment:
you can remove this function, because other than waiting for a specific message to trigger because your in a loop, you should never just wait...

If we need to wait then something is broken with the multi-tasking code...